prow/crier/reporters/resultstore/reporter_test.go

```diff
@@ -0,0 +1,218 @@
+						Org:  "org",
+						Repo: "repo",
+					},
+					DecorationConfig: &prowv1.DecorationConfig{
+						GCSConfiguration: &prowv1.GCSConfiguration{
+							Bucket: "bucket",
+						},
+					},
+				},
+				Status: prowv1.ProwJobStatus{
+					State:     prowv1.SuccessState,
+					StartTime: metav1.Time{Time: time.Date(2010, 10, 10, 18, 30, 0, 0, time.UTC)},
+					BuildID:   "build-id",
+				},
+			},
+			shouldReport: true,
+		},
+		{
+			name: "Report: false job not reported",
+			job: &prowv1.ProwJob{
+				Spec: prowv1.ProwJobSpec{
+					Type:   prowv1.PostsubmitJob,
+					Agent:  prowv1.KubernetesAgent,
+					Job:    "prow-job",
+					Report: false,
+					Refs: &prowv1.Refs{
+						Org:  "org",
+						Repo: "repo",
+					},
+					DecorationConfig: &prowv1.DecorationConfig{
+						GCSConfiguration: &prowv1.GCSConfiguration{
+							Bucket: "gs://bucket",
+						},
+					},
+				},
+				Status: prowv1.ProwJobStatus{
+					State:     prowv1.SuccessState,
+					StartTime: metav1.Time{Time: time.Date(2010, 10, 10, 18, 30, 0, 0, time.UTC)},
+					BuildID:   "build-id",
+				},
+			},
+			shouldReport: false,
+		},
+		{
+			name: "Non-GCS job not reported",
+			job: &prowv1.ProwJob{
+				Spec: prowv1.ProwJobSpec{
+					Type:   prowv1.PostsubmitJob,
+					Agent:  prowv1.KubernetesAgent,
+					Job:    "prow-job",
+					Report: true,
+					Refs: &prowv1.Refs{
+						Org:  "org",
+						Repo: "repo",
+					},
+					DecorationConfig: &prowv1.DecorationConfig{
+						GCSConfiguration: &prowv1.GCSConfiguration{
+							Bucket: "non-gcs://bucket",
+						},
+					},
+				},
+				Status: prowv1.ProwJobStatus{
+					State:     prowv1.SuccessState,
+					StartTime: metav1.Time{Time: time.Date(2010, 10, 10, 18, 30, 0, 0, time.UTC)},
+					BuildID:   "build-id",
+				},
+			},
+			shouldReport: false,
+		},
+		{
+			name: "Pending job not reported",
+			job: &prowv1.ProwJob{
+				Spec: prowv1.ProwJobSpec{
+					Type:   prowv1.PostsubmitJob,
+					Agent:  prowv1.KubernetesAgent,
+					Job:    "prow-job",
+					Report: true,
+					Refs: &prowv1.Refs{
+						Org:  "org",
+						Repo: "repo",
+					},
+					DecorationConfig: &prowv1.DecorationConfig{
+						GCSConfiguration: &prowv1.GCSConfiguration{
+							Bucket: "gs://bucket",
+						},
+					},
+				},
+				Status: prowv1.ProwJobStatus{
+					State:     prowv1.PendingState,
+					StartTime: metav1.Time{Time: time.Date(2010, 10, 10, 18, 30, 0, 0, time.UTC)},
+					BuildID:   "build-id",
+				},
+			},
+			shouldReport: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			gr := New(fakeConfigGetter{}.Config)
+			result := gr.ShouldReport(context.Background(), logrus.NewEntry(logrus.StandardLogger()), tc.job)
+			if result != tc.shouldReport {
+				t.Errorf("ShouldReport() got %v, want %v", result, tc.shouldReport)
+			}
+		})
+	}
+}
+
+type fakeConfigGetter struct {
+	c config.Config
+}
+
+func (fcg fakeConfigGetter) Config() *config.Config {
+	return &fcg.c
+}
```

Comment:
Nit: Maybe put this stuff up top to make it easier to discover?

Reply:
Done.